repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasEntityStoreV2.java

```diff
@@ -1802,6 +1814,46 @@ private AtlasStruct getStarredDetailsStruct(String assetStarredBy, long assetSta
         return starredDetails;
     }
 
+    private void createQualifiedNameHierarchyField(AtlasEntity entity, AtlasVertex vertex) {
+        MetricRecorder metric = RequestContext.get().startMetricRecord("createQualifiedNameHierarchyField");
+        boolean isDataMeshType = entity.getTypeName().equals(DATA_PRODUCT_ENTITY_TYPE) || entity.getTypeName().equals(DATA_DOMAIN_ENTITY_TYPE);
+        int qualifiedNameOffset = isDataMeshType ? 2 : 1;
+        try {
+            if (vertex == null) {
+                vertex = AtlasGraphUtilsV2.findByGuid(graph, entity.getGuid());
+            }
+            if (entity.hasAttribute(QUALIFIED_NAME)) {
+                String qualifiedName = (String) entity.getAttribute(QUALIFIED_NAME);
+                if (StringUtils.isNotEmpty(qualifiedName)) {
+                    vertex.removeProperty(QUALIFIED_NAME_HIERARCHY_PROPERTY_KEY);
+                    String[] parts = qualifiedName.split("/");
+                    StringBuilder currentPath = new StringBuilder();
+
+                    for (int i = 0; i < parts.length; i++) {
+                        String part = parts[i];
+                        if (StringUtils.isNotEmpty(part)) {
+                            if (i > 0) {
+                                currentPath.append("/");
+                            }
```

Comment:
Minor suggestion: 
This If can be avoided by moving `currentPath.append("/");` to the very end of the block  (inside)
`if (StringUtils.isNotEmpty(part)) {`

Reply:
e.g.
```
if (StringUtils.isNotEmpty(part))
    currentPath.append(part);
    // i>1 reason: we don't want to add the first part of the qualifiedName as it is the entity name
    // Example qualifiedName : default/snowflake/123/db_name we only want `default/snowflake/123` and `default/snowflake/123/db_name`
    if (i > qualifiedNameOffset) {
        if (isDataMeshType && (part.equals("domain") || part.equals("product"))) {
            continue;
        }
        AtlasGraphUtilsV2.addEncodedProperty(vertex, QUALIFIED_NAME_HIERARCHY_PROPERTY_KEY, currentPath.toString());
    }
    currentPath.append("/");
}
```

Reply:
Its there if there is no `/` in the qName then the split would be of len 1, and we will basically just not add `/` which will not be handled in this case. This condition makes sure it doesn't add unnecessary `/` after qualifiedName of glossary like assets.

Reply:
All we are doing is `i > 0` before `currentPath.append(part);`
Above suggestion results into exact same output & avoids `i>0` check many times in case QN is longer & with many `/`, but anyway this is not a blocker to merge